src/app/components/calendar/calendarComponent.tsx

```diff
@@ -1,25 +1,34 @@
 import Calendar from 'react-calendar';
 import 'react-calendar/dist/Calendar.css';
-import '../../globals.css'
-import { getWeekNumber } from '../../../../node_modules/react-calendar/dist/cjs/shared/dates.js';
+import '../../globals.css';
+import {getWeekNumber} from '../../../../node_modules/react-calendar/dist/cjs/shared/dates.js';
 
 type ValuePiece = Date | null;
 
 type Value = ValuePiece | [ValuePiece, ValuePiece];
 
-const CalendarComponent = ({ setCalendarHandle }: { setCalendarHandle: any }) => {
+const CalendarComponent = ({
+  setCalendarHandle,
+}: {
+  setCalendarHandle: (e: any) => void;
+}) => {
   const handleWeekNumber = (wN: number, date: Date) => {
-    const dayNumber =  date.getDate()
+    const dayNumber = date.getDate();
+
+    setCalendarHandle({target: {name: 'weekNumber', value: wN}});
+    setCalendarHandle({
+      target: {
+        name: 'weekDayNumbers',
+        value: Array.from({length: 7}, (_, i) => dayNumber + i),
+      },
+    });
+    setCalendarHandle({target: {name: 'year', value: date.getFullYear()}});
+  };
 
-    setCalendarHandle({ target: { name: 'weekNumber', value: wN }})
-    setCalendarHandle({ target: { name: 'weekDayNumbers', value: Array.from({length: 7}, (_, i) => dayNumber + i) }})
-    setCalendarHandle({ target: { name: 'year', value: date.getFullYear() }})
-  }
-  
   const handleMonth = (date: Date | Value) => {
```

Comment:
Unrelated but Date is a Value so this typing is confusing. Also Value is a pretty generic name, which leaves me guessing.

```suggestion
  const handleMonth = (date: Value) => {
```

Edit: ah, so it's actually from the Calendar library. Since imports do not work because of the way the library is set up (authors mistake ts ts), this would also be a way to access the library's definition, but in this particular case, I'd say it's not worth the complexity:

```tsx
import { NonUndefined } from 'react-hook-form';

// Value is the type of the first parameter of the onChange function of the Calendar component.
type CalendarOnChange = NonUndefined<React.ComponentProps<typeof Calendar>['onChange']>;
type Value = Parameters<CalendarOnChange>[0];
```

Maybe I'd just go with

```tsx
  // The library's typing also includes non-dates, but this seems never to happen.
  const handleMonth = (date: unknown) => {
    if (!(date instanceof Date) return;
    
    ...
  }
```

Wdyt? Leaving the original Value type is also an option

Reply:
yeah, my choice of props selection `Date | Value` is based on the library which is used for calendar component. I personally like the 2nd option as it looks very clear and pretty straightforward. Thank you. ❤️